p2p/node/helpers.go

```diff
@@ -66,26 +66,12 @@ func GenerateRandomNodesData(n int) []Node {
 	return res
 }
 
-// CheckUserPort tries to listen on a port and check whether is usable or not
-func CheckUserPort(port uint32) bool {
-	address := fmt.Sprintf("0.0.0.0:%d", port)
-	l, e := net.Listen("tcp", address)
-	if e != nil {
-		return true
-	}
-	l.Close()
-	return false
-}
-
 // GetUnboundedPort returns a port that is for sure unbounded or an error.
 func GetUnboundedPort() (int, error) {
-	port := crypto.GetRandomUserPort()
-	retryCount := 0
-	for used := true; used && retryCount < 10; used, retryCount = CheckUserPort(port), retryCount+1 {
-		port = crypto.GetRandomUserPort()
-	}
-	if retryCount >= 10 {
-		return 0, errors.New("failed to establish network, probably no internet connection")
+	l, e := net.Listen("tcp", ":0")
+	if e != nil {
+		return 0, e
 	}
-	return int(port), nil
+	defer l.Close()
```

Comment:
put defer after net.listen

Reply:
if `Listen` fails `l` will be nil so no need to `Close()`